notebook/notebookapp.py

```diff
@@ -539,6 +540,20 @@ def start(self):
     _("Allow the notebook to be run from root user.")
 )
 
+flags['use_redirect_file']=(
+    {'NotebookApp' : {'use_redirect_file' : True}},
+    """Disable launching browser by redirect file
+
+     For some environments (like Windows Subsystem for Linux (WSL) and Chromebooks),
+     launching a browser using a redirect file (introduced in notebook versions 
+     > 5.7.2) can lead the browser failing to load. This is because of the 
+     difference in file structures/paths between the runtime and the browser. 
+     
+     Disabling this setting will  by setting to False will disable this behavior, 
+     allowing the browser to launch by using a URL and visible token (as before).
+     """
```

Comment:
The redirect file was adopted as a security measure: without it, it's possible for another user on the same computer to steal your Jupyter session and run code as you. So the help about this option should be clear about that risk. Undoubtedly some people will turn it off without reading the help, but we can't do anything about that.